groclient/lib.py

```diff
@@ -403,20 +407,35 @@ def get_source_ranking(access_token, api_host, series):
     return get_data(url, headers, params).json()
 
 
-def rank_series_by_source(access_token, api_host, series_list):
-    for series in series_list:
+def rank_series_by_source(access_token, api_host, selections_list):
+    series_map = OrderedDict()
+    for selection in selections_list:
+        series_key = '.'.join([str(selection.get(type_id))
+                               for type_id in DATA_SERIES_UNIQUE_TYPES_ID
+                               if type_id != 'source_id'])
+        if series_key not in series_map:
+            series_map[series_key] = {}
+        elif None in series_map[series_key]:
+            continue
+        series_map[series_key][selection.get('source_id')] = selection
```

Comment:
Why do we need to key by series_key here ? It seems like we will always end up with one single series_key.

Reply:
The `selections_list` param can be many different series.

In your test example, you specified all of metric/item/region/frequency, only leaving out source id, so you only had one series key. But if I wanted to, I could put in two totally different series, like:

```py
rank_series_by_source([PRODUCTION_CORN_US, PRECIP_RAINFALL_CHINA])
```

and then the expected output would be something like:

1. production, corn, us, source 1
2. production, corn, us, source 2
3. precip, rainfall, china, source 1
4. precip, rainfall, china, source 2

It sorts by series, based on input order, first, and then by source ranking for each series.

I don't think that usage is very useful or intuitive, but it is valid currently, so it has to be maintained for backwards compatibility's sake.